src/Training.AirBnb.Clone.Backend/AirBnB.Api/Data/SeedDataExtensions.cs

```diff
@@ -148,6 +154,48 @@ private static async ValueTask SeedUsersAsync(this AppDbContext dbContext)
         );
 
         await appDbContext.ListingCategories.AddRangeAsync(listingCategories);
+    }
+
+    /// <summary>
+    /// Seeds listings into the AppDbContext from a JSON file.
+    /// </summary>
+    /// <param name="appDbContext"></param>
+    /// <param name="webHostEnvironment"></param>
+    /// <returns></returns>
+    private static async ValueTask SeedListingsAsync(this AppDbContext appDbContext, IHostEnvironment webHostEnvironment)
+    {
+        var randomDateTimeProvider = new RandomDateTimeProvider();
+        var listingsFileName = Path.Combine(webHostEnvironment.ContentRootPath, "Data", "SeedData", "Listings.json");
+        var users = await appDbContext.Users.Include(user => user.Role).Where(user => user.Role!.Type == RoleType.Host).ToListAsync();
+        var listingCategories = appDbContext.ListingCategories.Local.ToList();
+
+        // Retrieve listings
+        var listings = JsonConvert.DeserializeObject<List<Listing>>(await File.ReadAllTextAsync(listingsFileName))!;
+
+        var hostCounter = 0;
+
+        // Update listings
+        listings.ForEach(
+            listing =>
+            {
+                // Add random built date
+                listing.BuiltDate = DateOnly.FromDateTime(randomDateTimeProvider.Generate(new DateTime(1950, 1, 1), DateTime.Now.AddYears(3)));
```

Comment:
Is there a possibility that it will generate DateTime that is in the future?

Reply:
Exactly, I wanted to subtract years